qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -43,7 +43,7 @@ class IBMQBackend(BaseBackend):
     def __init__(
             self,
             configuration: BackendConfiguration,
-            provider,
+            provider: Union['AccountProvider', 'IBMQProvider'],
```

Comment:
Temporary forward reference marker. Needs to be dealt with after addressing the circular dependency that arises when the class is imported.